### ISSUE TITLE
Deepspeed on standalone app?

### ISSUE THREAD

**Issue by q5sys** [erew123/alltalk_tts#62]
This may not be possible since the docs only reference how to do it with the text-generation-ui.  
I already have libaio-devel installed (Im on Rocky Linux)
I've got alltalk_tts installed and its works just like it's supposed to. I'm now attempting to test it out with deepspeed.

Here's how I attempted to get it running on the standalone app.
1) I navigate to the directory that alltalk_tts is in.
2) I activate my conda environment. 
3) I run to `pip install deepspeed` 
4) I get the following error:

```
(alltalkenv) [q5@apollo alltalk_tts]$ pip install deepspeed
Collecting deepspeed
  Downloading deepspeed-0.12.6.tar.gz (1.2 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.2/1.2 MB 10.8 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error
  
  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [9 lines of output]
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "/tmp/pip-install-khqhn5sx/deepspeed_93b84e6a32f94f3495d451600d2ab13b/setup.py", line 100, in <module>
          cuda_major_ver, cuda_minor_ver = installed_cuda_version()
                                           ^^^^^^^^^^^^^^^^^^^^^^^^
        File "/tmp/pip-install-khqhn5sx/deepspeed_93b84e6a32f94f3495d451600d2ab13b/op_builder/builder.py", line 50, in installed_cuda_version
          raise MissingCUDAException("CUDA_HOME does not exist, unable to compile CUDA op(s)")
      op_builder.builder.MissingCUDAException: CUDA_HOME does not exist, unable to compile CUDA op(s)
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
```

You mentioned that textgen-ui overwrites the env variables, but since i'm using your native UI, will I still have that problem?
Since alltalk_tts works properly I wouldnt think there's an issue with the env variables for CUDA. 

Am I missing something obvious?

**Comment by erew123** [comment]
Hiya

DeepSpeed needs to be able to find the CUDA toolkit's nvcc to compile. DeepSpeed is not a pre-built bit of code, It actually compiles the code unique to your setup on Linux and it needs to be able to use `nvcc --version` to confirm the version of CUDA its compiling for. Once it has installed/compiled I **believe** you wont need to keep the Nvidia toolkit anymore (unless you reinstalled DeepSpeed).

So you would load your custom environment. You can use `pip show torch` if you need to confirm what version of CUDA you may have installed e.g. cu118 (cuda 11.8) or cu121 (cuda 12.1) etc

Either use your package manager to install the Nvidia CUDA toolkit or alternatively https://developer.nvidia.com/cuda-toolkit-archive

Once that's installed, confirm the path that CUDA is in... it may be /etc/alternative/cuda (wherever your Linux puts nvcc)

Load your python environment 

Set the environment path for CUDA_HOME e.g. `export CUDA_HOME=/usr/local/cuda-12.1/bin` or `export CUDA_HOME=/etc/alternative/cuda'

Then you should be able to type `nvcc --version` which should work and tell you the CUDA version. If it does, go ahead and `pip install deepspeed` and you are all sorted.

Remove the Nvidia toolkit (unless you want to keep it). Start AllTalk and you will need to activate DeepSpeed in the interface.

That's a loose guide.

**Comment by erew123** [comment]
Bar starting your Python environment, Steps 1, 3, 7 and 8 are the specific steps. (obviously matching the version of CUDA you have installed to the version of Pytorch).

https://github.com/erew123/alltalk_tts?tab=readme-ov-file#-linux-installation

**Comment by q5sys** [comment]
Thanks for the response. I'll poke around with it a bit tonight.  Assuming I dont run into any complexities introduced by my system, if I can get it working I'll put together a little text block about the steps I went through and file a PR to add that to the docs, so the next person who wants to do this has a clear section dedicated to the stand alone app.
Since I'm using your native UI, I wasn't sure how much I should rely on the instructions for someone using oobabooga.

**Comment by q5sys** [comment]
So I got it installed, interestingly I ran into a bit of an odd situation.
After setting the `env` when I tried to install deepspeed, I got the following error:
`FileNotFoundError: [Errno 2] No such file or directory: '/usr/local/cuda-12.1/bin/bin/nvcc'`

I dont know why it was doubling up the `bin` dir, but I figured instead of altering the path and potentially causing an issue for something else, I'd just set the path as part of `PATH` since I'll probably be using the CUDA toolkit for other things in the future. 

I'm not sure if this was just an oddity about my system (Rocky 9.3) or not.

In either case I got it working so I'm going to close this ticket. Thanks for the help.  PR filed with instructions submitted.
